pkg/registry/core/serviceaccount/strategy.go

```diff
@@ -76,9 +80,17 @@ func (strategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) fie
 
 // WarningsOnUpdate returns warnings for the given update.
 func (strategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
-	return nil
+	return warnIfHasEnforceMountableSecretsAnnotation(ctx, obj.(*api.ServiceAccount))
 }
 
 func (strategy) AllowUnconditionalUpdate() bool {
 	return true
 }
+
+func warnIfHasEnforceMountableSecretsAnnotation(ctx context.Context, serviceAccount *api.ServiceAccount) []string {
+	_, ok := serviceAccount.Annotations[sa.EnforceMountableSecretsAnnotation]
+	if ok {
+		return []string{fmt.Sprintf("deprecated annotation %q on service account %s/%s. %s", sa.EnforceMountableSecretsAnnotation, serviceAccount.Namespace, serviceAccount.Name, sa.WarningMessage)}
```

Comment:
the client already has access to the namespace and service account name, see https://github.com/kubernetes/kubernetes/blob/master/pkg/api/pod/warnings.go for examples of how succinct these usually are

```suggestion
		return []string{fmt.Sprintf("metadata.annotations[%s]: deprecated in v1.32+; prefer separate namespaces to isolate access to mounted secrets", sa.EnforceMountableSecretsAnnotation)}
```

Reply:
is there a lot more to say other than "use separate namespaces"? the current content at https://kubernetes.io/docs/concepts/configuration/secret/#configure-least-privilege-access-to-secrets doesn't seem that helpful, are we planning to update it to include a lot more info about using separate namespaces?

Reply:
Yes, I plan to push a PR to update https://kubernetes.io/docs/concepts/configuration/secret/#configure-least-privilege-access-to-secrets to mention the deprecation and suggest using separate namespaces for isolation. wdyt?

Reply:
the docs link is super long for a warning displayed on the CLI, and I'm not sure it actually adds that much additional info, let's omit it for now

Reply:
Ok. I will remove the doc link, namespace and service account name from admission warning as well and api docs.